src/gui/folderman.cpp

```diff
@@ -1588,25 +1589,40 @@ Folder *FolderMan::addFolder(AccountStatePtr accountStatePtr, const QString &loc
         folderDefinition.navigationPaneClsid = QUuid::createUuid();
 #endif
 
-    auto f = addFolder(accountStatePtr, folderDefinition);
+    auto finalize = [this, accountStatePtr, syncMode, localFolder](const FolderDefinition &folderDefinition, const QStringList &blacklist = {}) {
```

Comment:
I know lambdas are cool.

But in this case (for example) it is just not needed imho. The reasons are:
- The encapsulated code is just used once.
- The code flow is cluttered because you just move code from the place where it is used to the beginning of the method. Why?

For me, this makes things just more confusing.